tests/test_estimate_params.py

```diff
@@ -905,6 +905,188 @@ def test_incorrect_lens(self):
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = 1.123543297)
+        if not all(abs(m.parameters[key] - gt[key]) > 0.02 for key in keys):
+            self.fail("m.parameter shouldn't be too close to the original parameters")
+
+        # When tolerance is 
+        with self.assertRaises(ValueError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [1, 2, 3])
+
+        # 
+        with self.assertRaises(TypeError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [1, '2', '3'])
+        
+        with self.assertRaises(ValueError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = (1, 2, 3))
+        
+        # Using TNC
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys=keys, method = 'TNC', tol = 1e-9)
+        track1 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        # Default Values 
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = 1e-9)
+        track2 = m.calc_error(results.times, results.inputs, results.outputs)
+        
+        self.assertNotAlmostEqual(track1, track2)
+
+        # Reset parameters
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, 
+                          bounds=bound, keys=keys, method = 'TNC', tol = 1e-4)
+        track1 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        # So, for tol values between 1e-4 and bigger, we will continue to have the same results, whereas. 
+        # To determine it is 1e-34is the smallest, we have a tolerance check if it 1e-3 produces similar results, to which it does not.
+
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        
+        # Providing bounds here has some unwanted behavior. Provides worse predictions... Just a result of having tolerance
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, 
+                          bounds=bound, keys=keys, method='TNC', options = {'maxiter': 250}, tol=1e-9)
+        
+        track2 = m.calc_error(results.times, results.inputs, results.outputs)
```

Comment:
The reason for this test and exactly what we are doing is unclear to me. Can we edit the comment to be more descriptive?